simplesat/tests/test_repository.py

```diff
@@ -161,3 +166,31 @@ def test_find_packages(self):
 
         # Then
         self.assertEqual(len(packages), 0)
+
+    def test_contains(self):
+        # Given
+        packages_definition = textwrap.dedent(u"""\
+        dummy 1.0.1-1
+        dummy_with_appinst 1.0.0-1
+        dummy_with_entry_points 1.0.0-1
+        dummy_with_proxy 1.3.40-3
+        dummy_with_proxy_scripts 1.0.0-1
+        dummy_with_proxy_softlink 1.0.0-1
+        nose 1.2.1-1
+        nose 1.3.0-1
+        nose 1.3.0-2\
+        """)
+        packages = self.packages_from_definition(packages_definition)
+        repository = Repository(packages)
+
+        # When/Then
+        for package in repository:
```

Comment:
It might also be good to remake a package from scratch and check that contains still works to find the original. The implementation is intended to support that, right?